sktime/pipeline/tests/regression_tests/test_pipeline_regression.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+import pandas as pd
+
+from sktime.classification.distance_based import KNeighborsTimeSeriesClassifier
+from sktime.datasets import load_longley, load_arrow_head
+from sktime.forecasting.model_selection import temporal_train_test_split
+from sktime.forecasting.sarimax import SARIMAX
+from sktime.pipeline.pipeline import Pipeline
+from sktime.transformations.compose import Id
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+from sktime.transformations.series.difference import Differencer
+from sktime.transformations.series.exponent import ExponentTransformer
+from sktime.transformations.series.lag import Lag
+from sktime.utils._testing.hierarchical import _bottom_hier_datagen, _make_hierarchical
+
+
+def test_transformer_regression():
+    np.random.seed(42)
+    y, X = load_longley()
+    trafo_pipe = ExponentTransformer() * BoxCoxTransformer()
+    trafo_pipe.fit(X=X)
+    result = trafo_pipe.transform(X)
+    np.random.seed(42)
+    general_pipeline = Pipeline()
+    for step in [{"skobject" : ExponentTransformer(),
+                  "name":"exponent",
+                  "edges": {"X":"X"}},
+                 {"skobject" : BoxCoxTransformer(),
+                  "name":"BoxCOX",
+                  "edges": {"X":"exponent"}}]:
+        general_pipeline.add_step(**step)
+    general_pipeline.fit(X=X)
+    result_general = general_pipeline.transform(X)
+    pd.testing.assert_frame_equal(result, result_general)
+
+
+def test_classifier_regression():
+    np.random.seed(42)
+    X, y = load_arrow_head(split="train", return_X_y=True)
+    clf_pipe = ExponentTransformer() * KNeighborsTimeSeriesClassifier()
+    clf_pipe.fit(X, y)
+    result = clf_pipe.predict(X)
+    general_pipeline = Pipeline()
+    for step in [{"skobject" : ExponentTransformer(),
```

Comment:
note for later: we may like to think about simplified specification syntax, e.g.:

* if name not provided, generate a unique name
* if edges not provided, assume sensible defaults, i.e., last transformer given

further potential syntactic sugar:

* dunders `*` and `+` that work with the `Pipeline` and do the expected thing (`*` adds sequential after the last element added; `+` adds parallel after the last element's predecessor if unique)

I would do the above at the very last though. I think it is great for the user experience, but best done when the logic and structure internally are consolidated.